### PR TITLE
gh-2580 remove dynamic schema.

### LINKED ISSUE #2580: Fix FederatedStore checks for DYNAMIC_SCHEMA
Due to a partial implementation of #2008, currently FederatedOperationChains only validate the View if the graph does not have the `DYNAMIC_SCHEMA` trait:
https://github.com/gchq/Gaffer/blob/40bb337d2ca91fea2dadcd04cddec46021c169ee/store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/FederatedOperationChainValidator.java#L82
https://github.com/gchq/Gaffer/blob/40bb337d2ca91fea2dadcd04cddec46021c169ee/store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/FederatedStoreUtil.java#L134

This trait only exists in the FederatedStore as its getTraits() implementation returns ALL_TRAITS. This effectively means that only FederatedStores and ProxyStores which point to a FederatedStore will not be validated here. However, this cannot be used going forward because:
- FederatedStore fixes introduced in the FederatedStore alpha mean it will no longer return ALL_TRAITS
- the getTraits function is being replaced by the Operation which behaves differently, instead of returning ALL_TRAITS it will use the FederatedGetTraitsHandler which correctly returns the traits that the FederatedStore's stores have

Therefore, before the getTraits/getStoreTraits function can be completely removed, this usage has to be replaced (under this ticket). This can be done in various ways:
- deciding on a different solution to the partial implementation of #2008: either by fully implementing it or reverting it perhaps?
- replacing the FederatedStore check with something else smarter than this DYNAMIC_SCHEMA trait check

A solution to this must be discussed, decided on and implemented before getTraits can be completely removed and it seems that it will be best to wait until the other FederatedStore changes first to do so.

### PR DESCRIPTION
**Pull request by GCHQDev404**
# Related Issue

- Resolve #2580

### CODE REVIEW

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/FederatedStoreUtil.java [hunk 1]
```diff
@@ -146,10 +145,6 @@ public static <OP extends Operation> OP updateOperationForGraph(final OP operati
                     if (validView.hasGroups()) {
                         ((OperationView) resultOp).setView(validView);
                         // Deprecated function still in use due to Federated GetTraits bug with DYNAMIC_SCHEMA
```
Comment (t11947) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/FederatedStoreUtil.java:147]:
```suggestion
                        ((OperationView) resultOp).setView(validView);
```

**Review by t11947 — CHANGES_REQUESTED** [review]
Just a single suggestion removing something referencing DYNAMIC SCHEMA in a comment.

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/FederatedOperationChainValidator.java [hunk 2]
```diff
@@ -101,10 +100,7 @@ protected void validateViews(final Operation op, final User user, final Store st
                 if (graphIdValid) {
                     currentResult = new ValidationResult();
                     clonedOp.graphIdsCSV(graphId);
-                    // Deprecated function still in use due to Federated GetTraits bug with DYNAMIC_SCHEMA
-                    if (!graphSerialisable.getGraph().getStoreTraits().contains(StoreTrait.DYNAMIC_SCHEMA)) {
```
Comment (t92549) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/FederatedOperationChainValidator.java:105]: This only used to run on proxy stores and federated stores but now on all, is this okay?

Comment (GCHQDev404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/FederatedOperationChainValidator.java:105]: "**Not** (Traits contains Dynamic Schema)"

**Comment by GCHQDev404** [comment]
superseded

**Pull request closed without merging**